### ISSUE TITLE
Reproducibility of PD results (Adam et al 2024)

### ISSUE THREAD

**Issue by Supapoom** [ytanaka-bio/scIDST#1]
Dear authors,

Thank you for creating this amazing tool. I wonder how PD results (Adam et al 2024) could properly reproduce by scIDST method. Could you provide
1. The details of input data. It does not clear that how many features (as HVGs or whole transcriptome in datasets) and cells do you feed into **autoencoder.py**. Did you preprocess and annotate in Seurat before running Autoencoder?
2.  The tuned parameters of **autoencoder.py** and **classifier.py** to reduce the dimension if possible.
3.  The method of Differential Expression (DE) analysis between progressive stage and early stage cells in PD groups. Did you compare two groups of cell stage by aggregated cells (donor level) or individual cells (Supplementary Note 2)?

Thank you and looking forward to forthcoming response!

Best,

**Comment by ytanaka-bio** [comment]
Hello Supapoom

**>The details of input data. It does not clear that how many features (as HVGs or whole transcriptome in datasets) and cells do you feed into autoencoder.py. Did you preprocess and annotate in Seurat before running Autoencoder?**

Please use all features. Before running authoencoder, I recommend performing quality control in Seurat (e.g. removing low quality cells). I also performed cell annotation, but simply define cell type according to cell type specific markers.

**>The tuned parameters of autoencoder.py and classifier.py to reduce the dimension if possible.**

If you are fine, we can share the tuned model.

**>The method of Differential Expression (DE) analysis between progressive stage and early stage cells in PD groups. Did you compare two groups of cell stage by aggregated cells (donor level) or individual cells (Supplementary Note 2)?**

I believe that you mentioned about Fig. 1F and 1G. If so, yes. We separated individual cells into two groups with threshold 0.8 and performed DEG.

**Comment by Supapoom** [comment]
Dear authors,

Thank you very much for clarification!
Could you please share your tuned autoencoder and classifier model ?

**Comment by Supapoom** [comment]
My apology for this message, but could you share the tuned model (weights for autoencoder and tuned hyperparameters for classifier) if possible?

Best,

**Comment by ytanaka-bio** [comment]
Hello Supapoom
I apologize my delayed response.
Due to Ester holiday, I could not address this. 
I am preparing the models now, and will share them today or tomorrow.

**Comment by Supapoom** [comment]
My sincere apology for interrupting your vacation and thank you very much for your sharing.

**Comment by ytanaka-bio** [comment]
Can you access files from the following links:
https://udemontreal-my.sharepoint.com/:f:/g/personal/yoshiaki_tanaka_umontreal_ca/IgCrF_MiKa7cS7t1iIJMZ9uhAZoNO0B7eKiFvcf9Tki5r8M?e=6sNo4a

I am sorry that I could not find model files for classifier. Instead, we included examples of input files (count matrix and phenotype file)

**Comment by Supapoom** [comment]
Dear author,

Thank you for sharing the input and pretrained model files.

I've tried to train the classifier (with sigmoid output) using "disease" probabilistic labels from Reef/Snuba. The input of classifier.py is latent.csv (Autoencoder) and only single "disease" probabilistic label column. During the hyperparameter tuning, the val_accuracy shown "zero" for all trials, and the first trial was automatically selected as the best performances.

I am not sure that val_accuracy metric is still suitable for weakly classification. Probably, "Mean Absolute Error" seems reasonable for hyperparameter selection. Do you have opinions or suggestions for this issue?

**Comment by ytanaka-bio** [comment]
When we trained classifier, we used all three labels (disease, age, and sex). Can you try to train classifier by combining all labels?

**Comment by Supapoom** [comment]
Thank you very much for clarification. I will try to run the model on Adam et al. dataset again with all three labels.

However, is it always necessary to provide **_age_** and **_sex_** labels for classifier? Some datasets, such as Alzheimer's disease (Smith et al. 2022) do not have the _Young_ label for Reef/Snuba prediction. If 3 labels are required, the input of classifier in the **_age_** column will always be 1. Only _**disease**_ and _**sex**_ are able to turn into probabilistic labels from Reef/Snuba.

Here is an example of Reef/Snuba output, which will be the input for classifier.
[example.txt](https://github.com/user-attachments/files/26633317/example.txt)

If I misunderstood something about this, please let me know.

**Comment by ytanaka-bio** [comment]
It is not necessary to include "age" and other labels. But, in the analysis of Fig. 1, we used all labels.
Could you share the output of Reef/Snuba analysis?

**Comment by Supapoom** [comment]
Here are the Reef/Snuba results with [disease, age, sex] labels and only ['disease'] label with my observation so far.

[label_ws_vae_MIDBRAIN_ADAM2024_all_labels.csv](https://github.com/user-attachments/files/26652197/label_ws_vae_MIDBRAIN_ADAM2024_all_labels.csv)
[label_ws_vae_MIDBRAIN_ADAM2024_disease_label.csv](https://github.com/user-attachments/files/26652196/label_ws_vae_MIDBRAIN_ADAM2024_disease_label.csv)

During training with these 2 files, I received 2 different outputs of classifier.
1. If I train with 3 labels in the classifier, the model _could_ return  **Accuracy**, Here is my first trial of hyperparameter tuning.
> Trial 1 Complete [00h 13m 13s]
val_accuracy: 0.861106296380361
**Best val_accuracy So Far: 0.861106296380361**
Total elapsed time: 00h 13m 13s

2. If I train with only 'disease' labels in the classifier, the model _could not_ return **Accuracy** and it became "zero". Here is the first trial of hyperparameter tuning. 
> Trial 1 Complete [00h 13m 16s]
val_accuracy: 0.0
**Best val_accuracy So Far: 0.0**
Total elapsed time: 00h 13m 16s

In this case, the model will select **the first trial as the best hyperparameters**, because all trials return zero Accuracy. 
Here is my classifier setting. I use "Sigmoid" function as activation in the output dimension. 

`python3 classifier_analysis.py
    train
    -i "${LATENT_FILE}" 
    -p "${LABEL_FILE}"
    -o "${OUTPUT_PREFIX}"
    -t 32
    -e 20
    -x 100
    -b 1024
    -v 0.2
    -s 0.2
    -l`

For reproducibility, I agree to use 3 labels to reproduce the similar results in Fig 1. However, what's about the Reef/Snuba label of AD or Epilepsy dataset? Did you use "multiple" Reef/Snuba label for classifier? Because I'm also interested to use other PD datasets with only **disease status** for training.

**Comment by ytanaka-bio** [comment]
Thank you for the troubleshooting. It seems that K-ras tuner does not work well, when one disease label was used in the classifier. Our hypothesis is that gene expression is affected by multiple factors (disease, age, sex, and others) in a mutually dependent manner. Thus, we will include such clinical characteristics in the model training as much as possible. 

For AD data, we used disease state and sex (all samples are aged populations, so we did not use).
For epilepsy data, we used disease, age, and sex.

**Comment by Supapoom** [comment]
Dear author,
After training the 3 labels in the classifer with above settings, I've tried to reproduce the results by comparing the frequency of disease progressive cells in each cell types from **Supplementary Figure 2B**. Here is my figure's results.

<img width="400" height="400" alt="Image" src="https://github.com/user-attachments/assets/b7f89f4c-8649-43ff-b867-0e0a8e8d626a" />

<img width="400" height="400" alt="Image" src="https://github.com/user-attachments/assets/ea3c8ed1-6e58-49fb-a0a8-7a04c1272278" />

(**Left**: Supp Fig2B in Webhe _et al_)
(**Right**: Reproduced results. Purple; Disease Progressive cells, Cyan; Healthy/Early cells)

The frequency of disease progressive cells in Parkinson's disease Neurons shows greatly different from your original results, which affects downstream analysis (i.e. DEGs). I wonder about my cell type annotation maybe failed to generate the similar result that may affect a wrong cell composition. Here is my cell type marker and UMAP.

<img width="400" height="400" alt="Image" src="https://github.com/user-attachments/assets/a686dc87-7031-4a22-a495-1eff1d331aed" />

(Cell type markers)

<img width="400" height="400" alt="Image" src="https://github.com/user-attachments/assets/d7156932-efa3-400f-bcab-16842bec5adb" />

(UMAP)

I use SCANPY for data preprocessing (_log-normalizing, regressing mito genes, scaling and Harmony batch correction_) and cell type annotation using _sc.tl.rank_genes_groups()_. If my annotations are wrong, may I ask you for original _cell type annotation_ with cell-barcode file? Thank you very much for considering my request.

**Comment by ytanaka-bio** [comment]
Compared to our annotation, neuronal and T cell clusters seem to be larger in your annotation. Can you access to our Seurat Object from the following link?
https://udemontreal-my.sharepoint.com/:u:/g/personal/yoshiaki_tanaka_umontreal_ca/IQDS3yx2WYDLTIOgSKoIyI-_AYcCWmS9fLXfJ9D4YZpv_YQ?e=QWaqSV
Cell type information is in CellType slot (1=Neuron, 2=Oligodendrocyte, 3=OPC, 4=Astrocyte, 5=Endothelial cell, 6=Microglia, 7=Tcell)

**Comment by Supapoom** [comment]
Dear author,
After spending some times to investigate the model, I have some issues about the reproducibility of PD results.
1. Using your Autoencoder weight to infer into original dataset (matrix.tar), I got 500 latent representation. I assume that the latent values is consistent.
2. In Reef/Snuba, I ran the model twice with identical input (latent + binary files). The results are identical. I assume that the Reef output is consistent in your analysis.
3. In Classifer, I ran the model twice with 100 and 200 trials to get disease scores. The correlation between Reef output and both Classifer outputs are above 0.98. This mean that Reef and Classifer output have a similar trend (**Figure 1A, 1B**).
4. I applied your cell type annotation to investigate the Reef output of **_Neuronal cell type_** (**Figure 2A**). It shown that most of the neuron are "**Healthy**" (score below 0.8). This means if we train Healthy neuron in Classfier, the disease score should be nearly zero (**Figure 2B**). Still, the frequency of Disease neuron in my analysis is lower than your bar chart (**Supplementary Figure 2B**)

<img width="300" height="300" alt="Image" src="https://github.com/user-attachments/assets/53e24b54-d45d-4d6a-b853-db03903a7a6c" />
(Figure 1A) Reef output vs 100 trials Classifier output 

> 
<img width="300" height="300" alt="Image" src="https://github.com/user-attachments/assets/13ef340d-6040-49e5-876a-6c50e0508548" />
(Figure 1B) Reef output vs 200 trials Classifier output 

> 
<img width="1500" height="500" alt="Image" src="https://github.com/user-attachments/assets/990dfbc8-3a7d-43b6-bedf-f67d96e7fe73" />
(Figure 2A) Reef output (Neuron)

> 
<img width="1500" height="500" alt="Image" src="https://github.com/user-attachments/assets/ecf6dacd-25d4-4430-90ac-963abfd97f04" />
(Figure 2B) Classifier output (Neuron)

> 
Could you run your model again to confirm the consistence of your results?
If I am wrong or misunderstand some details, please let me know.

**Comment by ytanaka-bio** [comment]
Thank you for troubleshooting.

>  In Classifer, I ran the model twice with 100 and 200 trials to get disease scores. The correlation between Reef output and both Classifer outputs are above 0.98. This mean that Reef and Classifer output have a similar trend (Figure 1A, 1B).

Yes, Reef output should be similar with Classifer output, since the classifier is trained based on Reef output. There are several reasons why we used ANN classifier.
1- Reef output is relatively rough. ANN classifier can smooth the rough Reef values.
2-A major goal of scIDST is cross-dataset & cross-disease (like Fig. 2) analysis. 
The output of Reef and ANN classifier is also discussed in our Supplemental Note.

>  I applied your cell type annotation to investigate the Reef output of Neuronal cell type (Figure 2A). It shown that most of the neuron are "Healthy" (score below 0.8). This means if we train Healthy neuron in Classfier, the disease score should be nearly zero (Figure 2B). Still, the frequency of Disease neuron in my analysis is lower than your bar chart (Supplementary Figure 2B)

Could you provide your Classifier output? I will compare it with ours.

**Comment by Supapoom** [comment]
Here is my classifier output (200 trials). Thank you for further investigation.
[vae_MIDBRAIN_ADAM2024_train_ACCURACY_all_labels_ver2_result.csv](https://github.com/user-attachments/files/26950599/vae_MIDBRAIN_ADAM2024_train_ACCURACY_all_labels_ver2_result.csv)

**Comment by ytanaka-bio** [comment]
Hello Supapoom
Can you run reef_analysis.py by changing beta value (-b option)? For example, you can run it with "-b 0.3". Beta value is used to cutoff heuristics in Reef. Sometime, reef displayed unexpected error (maybe, due to no heuristics) and we changed beta value to solve it. In our re-analysis, "-b 0.3" works better.

<img width="323" height="406" alt="Image" src="https://github.com/user-attachments/assets/f977dfa1-1e80-464a-a663-de3c4a038680" />
<img width="319" height="400" alt="Image" src="https://github.com/user-attachments/assets/619407d4-89cb-4473-9a34-beccbc0e5eb4" />

**Comment by Supapoom** [comment]
Dear author,
When I ran Reef model using _**beta = 0.5**_, I did not receive any unexpected error. You can check the output file 1 **(output file 1; Figure 1)** in the attachment. Then I changed the **_beta to 0.3_**, but the Validation accuracy seem decreased **(output file 2; Figure 2)**. I have checked the Neuron Reef Label in both Beta parameter and it was inconsistent (Figure 3). 

I am not sure which beta is the better for what labels (I assumed that you used Beta = 0.3 for disease label, and Beta = 0.5 for age and sex label for Neural network input?). Could you please clarify why does beta=0.3 work better?

To improve the reproducibility and reliability of each figures and datasets, I would recommend to provide the input files (feature/ matrix/ barcode) and all tuned parameters (Autoencoder weights/parameters, Reef/Snuba parameters for each labels, Neural network weights/parameters). This will support your claim that the progressive cells are consistent and meaningful.

If I did any mistakes some steps or provided misinformation, please let me know.
Best regards,

**Output file 1** [reef_disease_beta05.txt](https://github.com/user-attachments/files/27035679/reef_disease_beta05.txt)
**Output file 2** [reef_disease_beta03.txt](https://github.com/user-attachments/files/27035685/reef_disease_beta03.txt)

<img width="889" height="440" alt="Image" src="https://github.com/user-attachments/assets/7080428a-bdd9-415c-9189-71bfada5f9b8" />

(Figure 1) Validation Accuracy of Reef across 10 train/test split (beta = 0.5)

<img width="889" height="440" alt="Image" src="https://github.com/user-attachments/assets/dc541982-3918-45ed-89fc-2824be0e9242" />

(Figure 2) Validation Accuracy of Reef across 10 train/test split (beta = 0.3)

<img width="2667" height="1500" alt="Image" src="https://github.com/user-attachments/assets/f8d97096-eb0d-47c1-b2bd-c80f1ad283cc" />

(Figure 3)
